soteria-rust/lib/tree_block.ml

```diff
@@ -766,6 +807,36 @@ let zero_range ofs size t =
       in
       ((), to_opt tree)
 
+let protect ofs size tag tb t =
+  match t with
+  | None -> miss_no_fix ~msg:"protect on none" ()
+  | Some t ->
+      let++ (), tree =
+        let@ () = with_bound_check t (ofs +@ size) in
+        Tree.protect ofs size tag tb t.root
+      in
+      ((), to_opt tree)
+
+let unprotect ofs size tag tb t =
+  match t with
+  | None -> miss_no_fix ~msg:"protect on none" ()
+  | Some t ->
+      let++ (), tree =
+        let@ () = with_bound_check t (ofs +@ size) in
+        Tree.unprotect ofs size tag tb t.root
+      in
+      ((), to_opt tree)
+
+let tb_access ofs size tag tb t =
+  match t with
+  | None -> miss_no_fix ~msg:"protect on none" ()
+  | Some t ->
+      let++ (), tree =
+        let@ () = with_bound_check t (ofs +@ size) in
+        Tree.tb_access ofs size tag tb t.root
+      in
+      ((), to_opt tree)
```

Comment:
- You forgot to fix the message in each copy-pasted line
- None-check + bound check can be factored out into a single function